crates/core/transaction/src/view.rs

```diff
@@ -94,6 +97,48 @@ impl TransactionView {
     pub fn action_views(&self) -> impl Iterator<Item = &ActionView> {
         self.body_view.action_views.iter()
     }
+
+    pub fn summary(&self) -> TransactionSummary {
+        let mut effects = Vec::new();
+
+        for action_view in &self.body_view.action_views {
```

Comment:
comment: Iterates through the visible action views and adds the contributing balances to the transaction's effects. Sanity check the required and provided balances.

Reply:
emulated logic from https://github.com/penumbra-zone/penumbra/blob/main/crates/core/transaction/src/plan/action.rs#L201 and https://protocol.penumbra.zone/main/transactions/actions.html

Reply:
cc @erwanor @cronokirby

Reply:
Looks OK to me